backend/pkg/exporter/modules/execution_deposits_exporter.go

```diff
@@ -163,59 +165,67 @@ func (d *executionDepositsExporter) OnFinalizedCheckpoint(event *constypes.Stand
 }
 
 func (d *executionDepositsExporter) OnHead(event *constypes.StandardEventHeadResponse) (err error) {
-	var nearestELBlock sql.NullInt64
-	err = db.ReaderDb.Get(&nearestELBlock, "select exec_block_number from blocks where exec_block_number > 0 order by slot desc limit 1")
-	if err != nil {
-		return err
-	}
-	if !nearestELBlock.Valid {
-		return fmt.Errorf("no block found for head slot %v", event.Slot)
-	}
-	log.Debugf("setting nearestELBlock for execution_deposits_exporter: %v", nearestELBlock.Int64)
-	d.CurrentHeadBlock.Store(uint64(nearestELBlock.Int64))
-	return nil
+	return nil // nop
 }
 
 func (d *executionDepositsExporter) exportLoop() {
 	ticker := time.NewTicker(time.Second * 10)
 	defer ticker.Stop()
 	for ; true; <-ticker.C {
-		currentHeadBlock := d.CurrentHeadBlock.Load()
-		if currentHeadBlock == 0 {
-			continue
-		}
-		if currentHeadBlock <= d.LastExportedBlock {
-			continue
-		}
-		err := d.exportTillBlock(currentHeadBlock)
+		err := d.export()
 		if err != nil {
-			log.Error(err, fmt.Sprintf("error during export to currentHeadBlock %d", currentHeadBlock), 0)
+			log.Error(err, "error during export", 0)
+			services.ReportStatus("execution_deposits_exporter", err.Error(), nil)
+		} else {
+			services.ReportStatus("execution_deposits_exporter", "Running", nil)
 		}
 	}
 }
 
-func (d *executionDepositsExporter) exportTillBlock(block uint64) (err error) {
-	// important: have to fetch the actual finalized epoch because even tho its called on finalized checkpoint it actually emits for each justified epoch
-	res, err := d.CL.GetFinalityCheckpoints("head")
+func (d *executionDepositsExporter) export() (err error) {
+	var headBlock, finBlock uint64
+	var g errgroup.Group
+	g.Go(func() error {
+		headSlot, err := d.CL.GetSlot("head")
+		if err != nil {
+			return fmt.Errorf("error getting head-slot: %w", err)
+		}
+		headBlock = headSlot.Data.Message.Body.ExecutionPayload.BlockNumber
+		return nil
+	})
+	g.Go(func() error {
+		finSlot, err := d.CL.GetSlot("finalized")
+		if err != nil {
+			return fmt.Errorf("error getting finalized-slot: %w", err)
+		}
+		finBlock = finSlot.Data.Message.Body.ExecutionPayload.BlockNumber
+		return nil
+	})
```

Comment:
this could be done in `onHead` & `onFinalizedCheckpoint` to follow the event driven approach

but since `export()` is in a fixed loop i would not bother with it rn since it would yield no benefit